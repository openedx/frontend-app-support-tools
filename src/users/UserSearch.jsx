import React, {
  useRef, useCallback,
} from 'react';
import PropTypes from 'prop-types';

import { Button, Form } from '@edx/paragon';

export default function UserSearch({ userIdentifier, searchHandler }) {
  const searchRef = useRef();

  const submit = useCallback((event) => {
    searchHandler(searchRef.current.value);
    event.preventDefault();
    return false;
  });

  return (
    <section className="mb-3">
      <Form className="m-0 row">
        <Form.Label className="mt-2" htmlFor="userIdentifier">Username, Email or LMS User ID</Form.Label>
        <Form.Control
          as="input"
          ref={searchRef}
          className="flex-grow-1 ml-1 mr-1"
          name="userIdentifier"
          defaultValue={userIdentifier}
        />

        <Button type="submit" onClick={submit} variant="primary">Search</Button>
      </Form>

    </section>
  );
}

UserSearch.propTypes = {
  userIdentifier: PropTypes.string,
  searchHandler: PropTypes.func.isRequired,
};

UserSearch.defaultProps = {
  userIdentifier: '',
};
