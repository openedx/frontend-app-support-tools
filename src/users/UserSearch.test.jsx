import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import UserSearch from './UserSearch';

describe('User Search Page', () => {
  let props;

  describe('renders correctly', () => {
    it('with correct user identifier', async () => {
      props = { userIdentifier: '', searchHandler: jest.fn() };
      render(<UserSearch {...props} />);
      const userSearchInput = await screen.findByTestId('userSearchInput');
      expect(userSearchInput.defaultValue).toEqual(props.userIdentifier);
    });
    it('with correct default user identifier', async () => {
      props = { searchHandler: jest.fn() };
      render(<UserSearch {...props} />);
      const userSearchInput = await screen.findByTestId('userSearchInput');
      expect(userSearchInput.defaultValue).toEqual('');
    });
    it('with submit button', async () => {
      props = { userIdentifier: '', searchHandler: jest.fn() };
      render(<UserSearch {...props} />);
      const submitButton = await screen.findByTestId('userSearchSubmitButton');
      expect(submitButton).toBeInTheDocument();
      expect(submitButton.textContent).toEqual('Search');
    });

    it('when submit button is clicked', async () => {
      const searchProps = { userIdentifier: 'staff', searchHandler: jest.fn() };
      render(<UserSearch {...searchProps} />);

      const submitButton = await screen.findByTestId('userSearchSubmitButton');
      fireEvent.click(submitButton);

      expect(searchProps.searchHandler).toHaveBeenCalledWith(
        searchProps.userIdentifier,
      );
    });

    it('matches snapshot', () => {
      props = { userIdentifier: '', searchHandler: jest.fn() };
      const { container } = render(<UserSearch {...props} />);
      expect(container).toMatchSnapshot();
    });
  });
});
