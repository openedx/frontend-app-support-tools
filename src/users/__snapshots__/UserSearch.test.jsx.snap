// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Search Page renders correctly matches snapshot 1`] = `
<section
  className="mb-3"
>
  <form
    className="m-0 row"
  >
    <label
      className="pgn__form-label mt-2"
      htmlFor="userIdentifier"
    >
      Username, Email or LMS User ID
    </label>
    <div
      className="pgn__form-control-decorator-group flex-grow-1 ml-1 mr-1"
    >
      <input
        className="form-control"
        defaultValue=""
        name="userIdentifier"
        onBlur={[Function]}
        onChange={[Function]}
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Search
    </button>
  </form>
</section>
`;
